### PR TITLE
Missing annotations i21244

### PR DESCRIPTION
**Pull request by jnpacker**
* Add printcolumn HOSTING for HypershiftDeployment, This lets you know how the provisioning is going.
* Transfer known annotations from HypershiftDeployment to HostedCluster

### CODE REVIEW

**Comment by jnpacker** [comment]
@ianzhang366 @zhujian7 Any better ideas then the HASH for checking the Annotation is present?
I felt it was cheaper to loop through the annotations on the HypershiftDeployment, and check the Hash, cost ( < N ), vs always looping through the list of annotations, cost ( N ).

But I'm open to interpretation.

File: pkg/controllers/default_resources.go [hunk 1]
```diff
@@ -62,14 +62,43 @@ func ScaffoldHostedCluster(hyd *hypdeployment.HypershiftDeployment) *hyp.HostedC
...
+	"ClusterAPIProviderAWSImage":                true,
+	"ClusterAPIKubeVirtProviderImage":           true,
+	"ClusterAPIAgentProviderImage":              true,
+	"ClusterAPIAzureProviderImage":              true,
+	"ExternalDNSHostnameAnnotation":             true,
+}
+
+// Looping through annotations on HypershiftDeployment and checking against the MAP is the fastest
+func transferHostedClusterAnnotations(hdAnnotations map[string]string, hcAnnotations map[string]string) {
+	for a, val := range hdAnnotations {
```
Comment (ianzhang366) [inline pkg/controllers/default_resources.go:95]: would be possible that the hcAnnotation is nil... then we will panic here

Comment (jnpacker) [reply pkg/controllers/default_resources.go:95]: Yes. Good catch.

**Comment by ianzhang366** [comment]
Do you mean we do this?

```
func transferHostedClusterAnnotations(hdAnnotations map[string]string, hcAnnotations map[string]string) {
	for k, val := range checkHostedClusterAnnotations {
		if val, ok := hdAnnotations[k] {
			hcAnnotations[k] = val
		}
	}
}
```

In this way, we are using the `checkHostedClusterAnnotations` as an allowList and it's O(N) N is the len(checkHostedClusterAnnotations)

File: pkg/controllers/default_resources.go (hunk 1 above)
Comment (jnpacker) [reply pkg/controllers/default_resources.go:95]: Actually its not possible. This function is only called by the scaffold HostedCluster, and there is always the Hypershift reference annotation included.

**Comment by jnpacker** [comment]
Yes. @ianzhang366 that was the other option. But I figured the hypershiftdeployment.annotation list most of the time will be LESS then len(checkHostedClusterAnnotations) so using the map is quicker.

**Comment by jnpacker** [comment]
Issue ref: https://github.com/stolostron/backlog/issues/21244

**Pull request merged**